### PR TITLE
Adds info about RL recipes.

### LINKED ISSUE #117: Document Reinforcement Learning related recipes
Document reinforcement learning recipes.

### PR DESCRIPTION
**Pull request by RogerPallares**
Closes #117 

Adds info about recipes: rl_training, rl_hyperoptimize and rl_simulate.

### CODE REVIEW

**Review by erikr — CHANGES_REQUESTED** [review]
Great work Roger! This was clearly a big effort and will be super helpful for you and others in the future.

I have a few minor suggestions and questions below. Thanks!

File: Recipes.md [hunk 1]
```diff
@@ -365,3 +365,152 @@ Default name: `xref_file.csv`, with the cross-referenced files and patients. Mor
 
 ## `visualize`
 This recipe launches a web application to interactively visualize Bedmaster and EDW data. Documentation is [[ here on its own Wiki page | Visualize ]].
+
+## `rl_train`
+
+### Overview
+Reinforcement learning recipe. Train a policy on the desired environment using DQN algorithm.
+Updates a Q function based on a neural network the desired amount of episodes.
```
[the remark this answers is not in the document]
Comment (RogerPallares) [reply Recipes.md:373]: Yes, this is not very well written actually. May be: _The Q function is based on a neural network and trained over a desired number of episodes_. What I want to say basically is that the policy is a neural network and you can choose for how many episodes you can train it.

File: Recipes.md [hunk 2]
```diff
@@ -365,3 +365,152 @@ Default name: `xref_file.csv`, with the cross-referenced files and patients. Mor
...
+
+### Input
+1. `--save_dir`: Directory to save all outputs.
```
[the remark this answers is not in the document]
Comment (RogerPallares) [reply Recipes.md:376]: Actually you are right, I will change this arg to `--output_folder`

File: Recipes.md [hunk 3]
```diff
@@ -365,3 +365,152 @@ Default name: `xref_file.csv`, with the cross-referenced files and patients. Mor
...
+2. `--tables_dir`: Directory to all tables. The discrete action space for the cardiovascular (CV) model is found here.
+3. `--init_conditions`: Initial conditions for the CV model, introduce 6 volumes.
+4. `--environment`: Environment of the RL problem. Current options: cartpole, cv_model. You can create your own environment. 
+5. `--num_episodes`: Number of episodes to train the policy.
+6. `--max_episode_steps`: Number of steps per episode.
+7. `--patient_state`: Initial patient state (level of sickness). Possibilities: normal, mild, moderate and severe.
+8. `--randomization`: Randomization for domain randomization. Possibilities: uniform and triangular.
+9. `--render`: Bool argument, if set environment is rendered at every 10 episodes.
+10. `--save`: Bool argument, if set, RL observations as well as render plots are saved.
+11. `--save_name`: Name of saved files.
```
[the remark this answers is not in the document]
Comment (RogerPallares) [reply Recipes.md:386]: In this case it is the name you use to save the files, let's say you put `test24_400episodes`, then your policy will be: `test24_400episodes_policy.h5`, the reward: `test24_400episodes_reward.png`, etc.

File: Recipes.md [hunk 4]
```diff
@@ -365,3 +365,152 @@ Default name: `xref_file.csv`, with the cross-referenced files and patients. Mor
...
+8. `--policy_name`: Policy name to be used (`.h5`). If argument not set, simulation is performed without policy.
+9. `--plot_list_states`: List of states to plot. Possible states are: v_lv, v_cas, v_cvs, v_rv, v_cap, v_cvp, p_lv, p_cas, p_cvs, p_rv, p_cap, p_cvp and c_out.
+10. `--plot_list_params`: List of parameters to plot. Possible parameters are: Ees_rv, Ees_lv, Vo_rv, Vo_lv, Tes_rv, Tes_lv, tau_rv, tau_lv, A_rv, A_lv, B_rv, B_lv, Ra_pul, Ra_sys, Rc_pul, Rc_sys, Rv_pul, Rv_sys, Rt_pul, Rt_sys, Ca_pul, Ca_sys, Cv_pul, Cv_sys, HR, Vt, Vs and Vu.
+11. `--plot_list_others`: List of other desired elements to plot. Available: actions, reward, pv_loop.
+12. `--render`: Bool argument, if plots are showed during simulation.
+13. `--save`: Bool argument, if set, all observations and parameters evolution during simulation are stored.
+14. `--save_name`: Name of saved files.
+
+### Output
+1. Observations, actions, rewards (`.npy` file) and parameters (`.pkl` file) over all the simulation (only if `--save` argument is set).
```
[the remark this answers is not in the document]
Comment (RogerPallares) [reply Recipes.md:495]: Maybe better if I say: Evolution of observations, actions, rewards and parameters over all the simulation?
How would you say that this output is the evolution of these elements simulated?

File: Recipes.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (erikr) [reply Recipes.md:386]: I see, so perhaps `--result_files_prefix` is a good name?

**Review by erikr — APPROVED** [review]

**Pull request merged**